src/serde/de.rs

```diff
@@ -0,0 +1,997 @@
+        self.reader.next()?;
+        let result = visitor.visit_seq(&mut *self);
+        self.reader.next()?;
+        result
+    }
+
+    fn deserialize_tuple_struct<V>(
+        self,
+        _name: &'static str,
+        _len: usize,
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        self.reader.step_in()?;
+        self.reader.next()?;
+        let result = visitor.visit_seq(&mut *self);
+        self.reader.next()?;
+        result
+    }
+
+    fn deserialize_map<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        self.reader.step_in()?;
+        self.reader.next()?;
+        let result = visitor.visit_map(&mut *self);
+        self.reader.next()?;
+        result
+    }
+
+    fn deserialize_struct<V>(
+        self,
+        name: &'static str,
+        fields: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        if name == "$datetime" && fields == ["$datetime"] {
+            let result = visitor.visit_map(TimestampDeserializer {
+                visited: false,
+                deserializer: self,
+            });
+            self.reader.next()?;
+            return result;
+        }
+
+        self.reader.step_in()?;
+        self.reader.next()?;
+        let result = visitor.visit_map(&mut *self)?;
+        self.reader.next()?;
+        Ok(result)
+    }
+
+    fn deserialize_enum<V>(
+        self,
+        _name: &'static str,
+        _variants: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        if let Some(ion_type) = self.reader.ion_type() {
+            match ion_type {
+                IonType::String => visitor.visit_enum(UnitVariantAccess::new(self)),
+                IonType::Struct => {
+                    self.reader.step_in()?;
+                    self.reader.next()?;
+                    let value = visitor.visit_enum(VariantAccess::new(self))?;
+                    self.reader.step_out()?;
+                    self.reader.next()?;
+                    Ok(value)
+                }
+                _ => Err(IonError::DecodingError {
+                    description: "expeted an enumeration".to_string(),
+                }),
+            }
+        } else {
+            Err(IonError::DecodingError {
+                description: "unexpected end of file".to_string(),
+            })
+        }
+    }
+
+    fn deserialize_identifier<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        let result = visitor.visit_string(self.reader.read_string()?);
+        self.reader.next()?;
+        result
+    }
+
+    fn deserialize_ignored_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        self.deserialize_any(visitor)
```

Comment:
Is `deserialize_ignored_any` supposed to just step over the current item in the stream? If so, why not just call `visitor.reader.next()` instead of doing serialization logic? (I'm just going off the method name.)